heartbeater.go

```diff
@@ -112,13 +111,24 @@ func (h *workerPoolHeartbeater) removeHeartbeat() {
 	conn := h.pool.Get()
 	defer conn.Close()
 
+	jobs := strings.Split(h.jobNames, ",")
+	inProgQueues := make([]interface{}, len(jobs))
 	workerPoolsKey := redisKeyWorkerPools(h.namespace)
 	heartbeatKey := redisKeyHeartbeat(h.namespace, h.workerPoolID)
 
-	conn.Send("SREM", workerPoolsKey, h.workerPoolID)
-	conn.Send("DEL", heartbeatKey)
-
-	if err := conn.Flush(); err != nil {
-		logError("remove_heartbeat", err)
+	// before removing the heartbeat hash, check to see if there is any work currently in progress
+	// if there is work in progress, then we will let the reaper clean up the heartbeat on a subsequent Start()
+	for _, job := range(jobs) {
+		inProgQueues = append(inProgQueues, redisKeyJobsInProgress(h.namespace, h.workerPoolID, job))
+	}
+	if count, err := redis.Int(conn.Do("EXISTS", inProgQueues...)); err != nil {
```

Comment:
Don't need to have this logic b/c we will be waiting for jobs to finish by the time we get here (heartbeater.stop() is called synchronously after worker jobs have completed).